piksi_multi_cpp/src/receiver/receiver.cc

```diff
@@ -14,8 +15,19 @@ Receiver::Receiver(const ros::NodeHandle& nh, const Device::Ptr& device)
   // Initialize SBP state.
   state_ = std::make_shared<sbp_state_t>();
   sbp_state_init(state_.get());
+
   // Register all relay callbacks.
   relay_cbs_ = SBPCallbackHandlerFactory::createAllSBPMessageRelays(nh, state_);
+
+  // Create observation callbacks
+  obs_cbs_ = std::make_unique<SBPObservationCallbackHandler>(nh, state_);
+
+  if (/*write to file*/ 1) {
```

Comment:
Regarding your comment on ROS params. Would you read the ROS parameter at this point directly? I think it's fine here. I guess if we always have objects that fulfill a certain task its fine to read the parameter directly from the server when the object is created.

Reply:
I think so, at least for the receiver(s) class its quite ok. just would try to keep it in one "entry class" to detangle ros a bit